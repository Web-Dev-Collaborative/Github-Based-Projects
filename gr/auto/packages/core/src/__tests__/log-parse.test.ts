import LogParse, { parsePR, parseSquashPR } from '../log-parse';

import makeCommitFromMsg from './make-commit-from-msg';

describe('parsePR', () => {
  test('should do nothing without merge commit', () => {
    const commit = makeCommitFromMsg('Not a merge');
    expect(parsePR(commit)).toStrictEqual(commit);
  });

  test('should parse number base and comment', () => {
    const commit = makeCommitFromMsg(
      'Merge pull request #1234 from Another PR\nComments about the PR'
    );

    expect(parsePR(commit)).toStrictEqual({
      ...commit,
      pullRequest: {
        number: 1234,
        base: 'Another PR'
      },
      subject: 'Comments about the PR'
    });
  });
});

describe('parseSquashPR', () => {
  test('should do nothing squash merge commit', () => {
    const commit = makeCommitFromMsg('Not a squash');
    expect(parseSquashPR(commit)).toStrictEqual(commit);
  });

  test('should parse PR number and comment', () => {
    const commit = makeCommitFromMsg('Some Message (#1234)');
    expect(parseSquashPR(commit)).toStrictEqual({
      ...commit,
      pullRequest: {
        number: 1234
      },
      subject: 'Some Message'
    });
  });
});

describe('normalizeCommits', () => {
  test('should do nothing with normal commits', async () => {
    const logParse = new LogParse();
    const commits = [
      makeCommitFromMsg('First'),
      makeCommitFromMsg('Second'),
      makeCommitFromMsg('Third')
    ];

    expect(await logParse.normalizeCommits(commits)).toMatchSnapshot();
  });
});
