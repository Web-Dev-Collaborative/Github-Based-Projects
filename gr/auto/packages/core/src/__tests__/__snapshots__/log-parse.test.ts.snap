// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeCommits should do nothing with normal commits 1`] = `
Array [
  Object {
    "authorEmail": "adam@dierkens.com",
    "authorName": "Adam Dierkens",
    "authors": Array [
      Object {
        "email": "adam@dierkens.com",
        "name": "Adam Dierkens",
      },
    ],
    "files": Array [],
    "hash": "foo",
    "labels": Array [],
    "pullRequest": undefined,
    "subject": "First",
  },
  Object {
    "authorEmail": "adam@dierkens.com",
    "authorName": "Adam Dierkens",
    "authors": Array [
      Object {
        "email": "adam@dierkens.com",
        "name": "Adam Dierkens",
      },
    ],
    "files": Array [],
    "hash": "foo",
    "labels": Array [],
    "pullRequest": undefined,
    "subject": "Second",
  },
  Object {
    "authorEmail": "adam@dierkens.com",
    "authorName": "Adam Dierkens",
    "authors": Array [
      Object {
        "email": "adam@dierkens.com",
        "name": "Adam Dierkens",
      },
    ],
    "files": Array [],
    "hash": "foo",
    "labels": Array [],
    "pullRequest": undefined,
    "subject": "Third",
  },
]
`;
